mypy/main.py

```diff
@@ -823,6 +823,8 @@ def add_invertible_flag(
         help="Make arguments prepended via Concatenate be truly positional-only",
         group=strictness_group,
     )
+    # Experiment flag to detect undefined variables being used.
+    add_invertible_flag("--disallow-undefined-vars", default=False, help=argparse.SUPPRESS)
```

Comment:
That is interesting! I don't have a strong opinion here. I think we'll want an error code eventually for `type: ignore[partially-defined]`. I'm guessing that if we have an error code, we shouldn't have a separate flag?

Reply:
> I think we'll want an error code eventually for `type: ignore[partially-defined]`

IMO it would definitely make sense for it to have a separate error-code. Reusing `misc` doesn't feel right.

> I'm guessing that if we have an error code, we shouldn't have a separate flag? So we should probably just make it an error code now. Is my understanding correct here?

Some error codes might also have flags but my understanding is that these are all legacy cases. The idea is to move more things to error-codes in the future.

Reply:
I agree that it would be better to have a dedicated error code for this (that is not enabled by default) and not have a dedicated flag. We'd only run the analysis if the error code is enabled.

Reply:
These error codes are nice :) Done!